### ISSUE TITLE
Think about versioning and recomputation

### ISSUE THREAD

**Issue by amosr** [ambiata/icicle#56]
When a program / virtual feature changes, we'll need to recompute the *whole* thing for that feature. Basically, clearing the snapshots and starting from scratch.

This is interesting when combined with fusion: when adding a new virtual feature on a concrete, we're actually changing the program that computes all virtuals on a given concrete, so all virtuals on that concrete need to be recomputed from scratch.

It may be surprising that removing a feature also requires recomputation: the fused program changes, of course, but because we are squashing down duplicate resumable folds, the unique names of the resumable folds can change. Damn!

In short, it looks like more or less adding, modifying or removing a virtual feature will require recomputing all virtual features attached to that concrete feature.
Does this sound like a problem?

**Comment by amosr** [comment]
We can sort this out later if it becomes a problem. Basically, when condensing queries together, we should be careful to keep all resumables instead of squashing them together. Or, load/save resumable could take a list of the feature name and so on, instead of just a single one.
